storm-core/src/jvm/backtype/storm/transactional/state/TransactionalState.java

```diff
@@ -59,7 +59,8 @@ protected TransactionalState(Map conf, String id, Map componentConf, String subr
                               .get(Config.TOPOLOGY_KRYO_REGISTER));
             }
             String transactionalRoot = (String)conf.get(Config.TRANSACTIONAL_ZOOKEEPER_ROOT);
-            String rootDir = transactionalRoot + "/" + id + "/" + subroot;
+            String topologyName = (String)conf.get(Config.TOPOLOGY_NAME);
```

Comment:
What if topologyName is null?
Should we give these configs default values？ 
Something like:
```
String topologyName = Utils.parseString(conf.get(Config.TOPOLOGY_NAME), "default-topology-name");
```